pandas/core/groupby/generic.py

```diff
@@ -898,22 +908,20 @@ def aggregate(self, func=None, *args, **kwargs):
 
     agg = aggregate
 
-    def _iterate_slices(self):
-        if self.axis == 0:
-            # kludge
-            if self._selection is None:
-                slice_axis = self.obj.columns
-            else:
-                slice_axis = self._selection_list
-            slicer = lambda x: self.obj[x]
+    def _iterate_slices(self) -> Iterator[Tuple[Hashable, Series]]:
+        obj = self._selected_obj.copy()
```

Comment:
why copy?

Reply:
Mistake on my end. Didn’t want to mutate passed frame in the axis=1 case but I can get rid of this

Reply:
Off topic: can we make the axis=1 case unnecessary by just transposing when we see it?  Not sure exactly how that would work, but it would let us get rid of some lambda-wrapping, which would be nice.